handlers/handler.go

```diff
@@ -135,3 +135,29 @@ func IndexS3Object(s3objectURL string) {
 
 	updateMetadataObjectWrapper(uuid, configInfo, mdsUploadedBody)
 }
+
+func CreateUUID(key string) (string, string, error) {
+	split_key := strings.Split(key, "/")
+	var uuid string
+	var filename string
+	// here uuid can be dg.XXXX/<uuid> or
+	// just <uuid> to distinguish between
+	// them and to keep the filename consistent
+	foundGUID, errGUID := id.Parse(split_key[0])
+	var foundPrefix id.UUID
+	var errPrefix error
+	if len(split_key) > 1 {
+		foundPrefix, errPrefix = id.Parse(split_key[1])
+	}
+
+	if errGUID == nil && foundGUID != id.Nil {
+		uuid = split_key[0]
+		filename = split_key[len(split_key)-1]
+	} else if errPrefix == nil && foundPrefix != id.Nil {
+		uuid = strings.Join(split_key[:2], "/")
+		filename = strings.Join(split_key[2:], "/")
+	} else {
+		return "", "", fmt.Errorf("Cannot process the UUID")
+	}
+	return uuid, filename, nil
+}
```

Comment:
What do you think about something like the following? This allows for cases in which prefixes contain a slash and cases in which there is no prefix but a subdirectory is included in the filename.
```
resolveUUID(key) {
  keyParts = strings.Split(key, "/")
  uuidIndex = -1
  for i, part in enumerate(keyParts) {
    foundUUID = id.Parse(part)
    if foundUUID != id.Nil {
      uuidIndex = i
      break
    }
  }
  if uuidIndex == -1 {
    return "", "", fmt.Error("...")
  }
  fullUUID = strings.Join(keyParts[:uuidIndex+1], "/")
  filename = strings.Join(keyParts[uuidIndex+1:], "/")
  return fullUUID, filename, nil
}
```